sdk/resourcemanager/Azure.ResourceManager/src/Custom/Resources/ProviderCollection.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.ResourceManager.Core;
+
+namespace Azure.ResourceManager.Resources
+{
+    /// <summary>
+    /// A class representing collection of resources and their operations over their parent.
+    /// </summary>
+    public partial class ProviderCollection : ArmCollection
```

Comment:
Does this type have to be public?

Reply:
yes, its an auto-generated [Resource]Client.  We are just added internal methods to it.